### ISSUE TITLE
Bot gives up on a trade after an error

### ISSUE THREAD

**Issue by Haarolean** [TF2Autobot/tf2autobot#273]
**Describe the bug**
Bot tried to confirm the trade, got an exception and gave up on this trade, hasn't tried to confirm anymore.

**To Reproduce**

1. Buy something
2. Get an error during confirming
3. profit

**Expected behavior**
Bot will reconfirm later in a retriable

**Screenshots and logs**
```
2021-01-13 15:33:50 ESC[32minfoESC[39m: Message sent to xxx: I have accepted your offer. The trade will be finalized shortly. If the trade does not finalize after 1-2 minutes has passed, please send your offer again, or add me and use the !sell/!sellcart or !buy/!buycart command.
2021-01-13 15:33:50 ESC[34mdebugESC[39m: Offer #4396768235 from xxx handler is done with offer
2021-01-13 15:33:50 ESC[34mdebugESC[39m: Connected to socket server
2021-01-13 15:33:51 ESC[35mtradeESC[39m: Offer #4396768235 from xxx successfully accepted; confirmation required
2021-01-13 15:33:51 ESC[34mdebugESC[39m: Accepting mobile confirmation... {"offerId":"4396768235"}
2021-01-13 15:33:51 ESC[34mdebugESC[39m: Offer #4396768235 from xxx done doing action on offer
2021-01-13 15:33:51 ESC[34mdebugESC[39m: Processing next offer
2021-01-13 15:33:51 ESC[34mdebugESC[39m: Already processing offer or queue is empty
2021-01-13 15:33:57 ESC[34mdebugESC[39m: Error while trying to accept mobile confirmation on offer #4396768235:  Could not act on confirmation {}
```
And this keeps going on forever:
`2021-01-13 15:35:57 ESC[36mverboseESC[39m: 1 incoming offer (0 on hold), 0 outgoing offers (0 on hold)`

Version: 2.3.5

**Comment by idinium96** [comment]
send your polldata.json file.

**Comment by Haarolean** [comment]
@idinium96 Sorry for the delay. I've sent you one via email since the file contains some private information like steamids and stats.

**Comment by idinium96** [comment]
Thanks.

This is actually a known issue, as you can see from the polldata.json, the offer is marked as accepted, that's why the bot won't bother to re-process the offer.

I am not sure how to solve this by myself. Will need help from others.

```
"4396768235": {
      "partner": "76561199084969980",
      "handleTimestamp": 1610552003769,
      "notify": true,
      "dict": {
        "our": {
          "184;6": {
            "amount": 1,
            "stock": 1,
            "maxStock": 1
          }
        },
        "their": {
          "5002;6": {
            "amount": 13,
            "stock": 13,
            "maxStock": 0
          }
        }
      },
      "value": {
        "our": {
          "total": 117,
          "keys": 0,
          "metal": 13
        },
        "their": {
          "total": 117,
          "keys": 0,
          "metal": 13
        },
        "rate": 52.33
      },
      "prices": {
        "184;6": {
          "buy": {
            "keys": 0,
            "metal": 1
          },
          "sell": {
            "keys": 0,
            "metal": 13
          }
        }
      },
      "handledByUs": true,
      "handleTime": 26389,
      "action": {
        "action": "accept",
        "reason": "VALID",
        "meta": {
          "highValue": {
            "has": {
              "our": true,
              "their": false
            },
            "items": {
              "our": {
                "skus": [
                  "184;6"
                ],
                "names": [
                  "Gentleman's Gatsby\n🎨 Painted: Pink as Hell"
                ]
              },
              "their": {
                "skus": [],
                "names": []
              }
            },
            "isMention": {
              "our": false,
              "their": false
            }
          }
        }
      },
      "actionTimestamp": 1610552030158,
      "actionTime": 1313,
      "actedOnConfirmation": true,
      "actedOnConfirmationTimestamp": 1610552031471,
      "confirmationTime": 5673,
      "finishTimestamp": 1610565130874,
      "isAccepted": true
    }
```

**Comment by Haarolean** [comment]
Can't we implement a retriable scheduler here to reconfirm failed offers later instead of just failing?
https://github.com/TF2Autobot/tf2autobot/blob/aac86e827c9709b841fa3f002e648743ce4d911b/src/classes/Trades.ts#L457

**Comment by idinium96** [comment]
@joekiller @Bonfire

**Comment by joekiller** [comment]
Sounds more like a bug/feature for steam-tradeoffer-manager.

Sent via mobile

**Comment by idinium96** [comment]
Implemented an alternative in [`v3.1.0` - `v3.1.2`](https://github.com/TF2Autobot/tf2autobot/releases). Closing.
